### ISSUE TITLE
Package listing like list-packages

### ISSUE THREAD

**Issue by Compro-Prasad** [raxod502/straight.el#290]
Is there a version of `list-packages` for straight.el?

**Comment by raxod502** [comment]
No, not currently. Personally I do not think that this interface is a very useful one; perhaps you really want something more specific. If you want a list of all available packages, you already get one via `M-x straight-use-package` or `M-x straight-get-recipe`, etc. If you want a list of registered packages in the current Emacs session, you could hack that together like this:

```elisp
(require 'subr-x)
(dolist (package (sort (hash-table-keys straight--recipe-cache)
                       #'string-lessp))
  (insert (format
           "%25s | %s\n"
           package (plist-get (gethash package straight--recipe-cache)
                              :local-repo))))
```

Which gives:

```
        aggressive-indent | aggressive-indent-mode
            all-the-icons | all-the-icons.el
            anaconda-mode | anaconda-mode
              apache-mode | apache-mode
              apples-mode | apples-mode
                    async | emacs-async
            atomic-chrome | atomic-chrome
                   auctex | auctex
                      avy | avy
                 bind-key | use-package
              buffer-move | buffer-move
                    cider | cider
                   cl-lib | elpa
```

Pull requests welcome.

**Comment by Compro-Prasad** [comment]
I actually browse new packages in the list which have a descent description and a link to their GitHub page.

**Comment by Compro-Prasad** [comment]
> Personally I do not think that this interface is a very useful one

Most of the operating systems have package listing feature with at least their description. I can at least expect a searching feature which searches through package as well as its description.

**Comment by igmacs** [comment]
I agree that there should be a package listing feature in Emacs, but I also agree that it doesn't necessarily need to be provided by straight, as long as straight lets you use other providers. The problem is that straight currently complains if you use the built-in one, `describe-package`.  In order for that to list all packages, you need to set the `package-archives` variable and call `(package-refresh-contents)` . The problem is that will create the `emacs.d/elpa` folder, which will result in a straight warning in the next startup, even if there isn't actually any package installed with `package.el`. In that case, the `emacs.d/elpa` folder has only two directories: `archives/` and `gnupg/`. Could a solution be to allow those two directories to exist when doing [this check](https://github.com/radian-software/straight.el/blob/4b6289f42a4da0c1bae694ba918b43c72daf0330/straight.el#L7475), and only display the warning if there are more directories than those inside `emacs.d/elpa`?

**Comment by raxod502** [comment]
Yeah, that would be ok, PR for that would definitely be merged.

**Comment by igmacs** [comment]
Here is one: https://github.com/radian-software/straight.el/pull/1279

WDYT? Let's discuss there from now on, as it is only partially related to this issue
